include/FastBVH/impl/Memory.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <FastBVH/impl/Types.h>
+
+#include <cstdlib>
+
+namespace FastBVH {
+
+namespace impl {
+
+//! Allocates memory on the device.
+//! \param size The number of bytes to allocate.
+//! \return A pointer to the allocated memory.
+inline void* allocateDeviceMemory(Size size) {
```

Comment:
This should probably be renamed. When I was trying to get a working CUDA build, this would allocate memory on the GPU when building with the Nvidia compiler.